core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -951,6 +951,11 @@ abstract class Record(private[chisel3] implicit val compileOptions: CompileOptio
             " See the cookbook entry 'How do I deal with the \"unable to clone\" error?' for more details."
         )
       }
+      if (field.binding.isDefined) {
```

Comment:
```suggestion
      if (clone.elements(name).binding.isDefined) {
```
This was checking the `field` of the parent, not the `clone`.

Although please assign `clone.elements(name)` to a `val` above rather than doing the lookup twice... it will probably be optimized by the compiler but I'd rather not risk it.

Reply:
Hmm I'm seeing that with `clone.elements(name)`, the `binding.isDefined` is always false and the test I added is failing with no exceptions

Reply:
did we wanna check the clone or the field itself? i thought the field was the thing that we shouldn't be cloning if it's bound already, not the clone